### PR TITLE
Add coverage for Python torch bindings

### PR DESCRIPTION
**Pull request by PicoCentauri**
Using the same logic and structure as https://github.com/lab-cosmo/metatensor/pull/381.

----
:books: Documentation preview :books:: https://rascaline--234.org.readthedocs.build/en/234/

### CODE REVIEW

**Comment by Luthaf** [comment]
It looks like the coverage is very confused... it is reporting empty lines as not-covered: https://app.codecov.io/github/Luthaf/rascaline/commit/ae4cb69ff05f267a229737158a11581baafc7e56/blob/python/rascaline-torch/rascaline/torch/__init__.py, just above `_load_library()`

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: .tox
-          key: tox-${{ matrix.os }}-${{ hashFiles('pyproject.toml', 'setup.cfg', 'tox.ini') }}
+          key: tox-${{ hashFiles('pyproject.toml', 'setup.cfg', 'tox.ini') }}
```
Comment (Luthaf) [inline .github/workflows/coverage.yml:50]: why remove matrix.os from the cache key?

File: tox.ini [hunk 2]
```diff
@@ -183,3 +186,19 @@ commands =
...
+
+[coverage:report]
+skip_covered = True
+show_missing = True
+omit =
+    tests/.*
+    examples/.*
+
+[coverage:paths]
+notorch =
```
Comment (Luthaf) [inline tox.ini:198]:
```suggestion
rascaline =
```

File: tox.ini [hunk 3]
```diff
@@ -183,3 +186,19 @@ commands =
...
+    python/rascaline/rascaline
+    .tox/*/lib/python*/site-packages/rascaline
+
+torch =
```
Comment (Luthaf) [inline tox.ini:202]:
```suggestion
rascaline_torch =
```

**Review by Luthaf — COMMENTED** [review]
looks good overall!

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (PicoCentauri) [reply .github/workflows/coverage.yml:50]: Because, we have no matrix here. We only run this on ubuntu 22

**Comment by Luthaf** [comment]
Seems that codecov is no longer adding a commit check with the overall coverage, we'll have to look into this. But this PR should be fine!

**Pull request merged**